api/src/main/java/io/strimzi/api/kafka/model/KafkaJmxOptions.java

```diff
@@ -24,6 +24,8 @@
 public class KafkaJmxOptions implements UnknownPropertyPreserving, Serializable {
     private static final long serialVersionUID = 1L;
     private KafkaJmxAuthentication authentication;
+    private JmxTransSpec jmxTransSpec;
```

Comment:
I haven't a strong opinion right now so you can easily convince me that it's the right way to go :-) but ... we are putting an entire deployment configuration (about JmxTrans) inside the Jmx options related part which just enable/disable jmx and configures auth right now. Maybe we could put the JmxTrans outside of it? @tombentley @scholzj what's your thoughts as well?

Reply:
I was wondering if users would enable JMX using the `jmxOptions` field but then they want to use their own mechanism (maybe an application running in a different pod to scrape metrics). In this case, JmxTrans is one of the possible mechanisms that Strimzi supports out of the box so I would put it outside of `jmxOptions`. But let's see what the others think first.

Reply:
TBH, my expectation was also that the JMX Trans deployment configuration would be on the same level as the other things such as Kafka Exporter are. Not inside the `spec.kafka` path.

Reply:
I think that it is reasonable to put it on the same level as Kafka Exporter. My proposed CR would be: 
```
apiVersion: kafka.strimzi.io/v1beta1
kind: Kafka
metadata:
  name: my-cluster
spec:
  kafka:
   /*
    *
  jmxTrans:
    outputDefinitions:
      - outputType: "com.googlecode.jmxtrans.model.output.StdOutWriter"
        name: "standardOut"
      - outputType: "com.googlecode.jmxtrans.model.output.GraphiteOutputWriter"
        host: "mylogstash.com"
        port: 31028
        flushDelay: 5
        name: "logstash"
    queries:
      KafkaQueries:
        - targetMBean: "kafka.server:type=BrokerTopicMetrics,name=*"
          attributes:  ["Count"]
          outputs: ["standardOut"]
      ZookeeperQueries (In the future):
        - targetMBean: "kafka.server:type=BrokerTopicMetrics,name=*"
          attributes:  ["Count"]
          outputs: ["standardOut"]
```
This would allow for future development!